localflavor/au/validators.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.core.exceptions import ValidationError
+from django.core.validators import RegexValidator
+from django.utils.translation import ugettext_lazy as _
+
+
+class AUBusinessNumberFieldValidator(RegexValidator):
+    """
+    Validation for Australian Business Numbers.
+
+    .. versionadded:: 1.3
+    """
+
+    error_message = _('Enter a valid ABN.')
+
+    def __init__(self):
+        eleven_digits_with_whitespace = '^(\s*\d\s*){11}$'
+        super(AUBusinessNumberFieldValidator, self).__init__(
+            regex=eleven_digits_with_whitespace, message=self.error_message)
+
+    def _is_valid(self, value):
+        """
+        Return whether the given value is a valid ABN.
+
+        See https://en.wikipedia.org/wiki/Australian_Business_Number
+        for a description of the validation algorithm.
```

Comment:
A link to the validation algorithm would be nice. I found this which seems useful:
http://www.clearwater.com.au/code